tokio-epoll-uring/src/system/lifecycle/handle.rs

```diff
@@ -171,4 +171,33 @@ impl crate::SystemHandle {
         let inner = self.inner.as_ref().unwrap();
         execute_op(op, inner.submit_side.weak(), None).await
     }
+
+    pub async fn statx<F: IoFd + Send>(
+        &self,
+        file: F,
+    ) -> (
+        F,
+        Result<
+            Box<uring_common::libc::statx>,
+            crate::system::submission::op_fut::Error<std::io::Error>,
+        >,
+    ) {
+        // TODO: avoid the allocation, or optimize using a slab cacke?
+        let buf: Box<uring_common::libc::statx> = Box::new(
+            // TODO replace with Box<MaybeUninit>, https://github.com/rust-lang/rust/issues/63291
+            // SAFETY: we only use the memory if the fstat succeeds, should be using MaybeUninit here.
+            unsafe { std::mem::zeroed() },
```

Comment:
Note the API which is unstable is `Box::uninit`, this is stable:

```suggestion
        let buf: Box<MaybeUninit<uring_common::libc::statx>> = Box::new(
            MaybeUninit::uninit()
```

But without `Box::assume_init` this would require extra wrestling.

Reply:
> extra wrestling

=  keeping track of the MaybeUninit<T> vs T state in the SubmittingBox.